src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java

```diff
@@ -399,6 +402,21 @@ private static DetailAST getLeftNode(DetailAST node) {
         return result;
     }
 
+    /**
+     * Ascends AST to determine if given node is part of a pattern
+     * definition.
+     *
+     * @param node the node to check
+     * @return true if node is in pattern definition
+     */
+    private static boolean isInPatternDefinition(DetailAST node) {
+        DetailAST parent = node.getParent();
+        while (parent != null && parent.getType() != TokenTypes.PATTERN_DEF) {
+            parent = parent.getParent();
```

Comment:
why not?
````
DetailAST parent = node;
do {
    parent = parent.getParent();
} while (parent != null && parent.getType() != TokenTypes.PATTERN_DEF);
````

Also we are passing through parents uncontrolled. It doesn't matter if we pass any other kind of def before hitting the pattern def?

Reply:
> why not...

Please explain the advantage of this; to me the name of variable declared (`parent`) and initialization (`node.getParent();`) makes sense. Since we are checking operands, there is absolutely no chance that `node` is null (it will always have an operator as a parent).

>  It doesn't matter if we pass any other kind of def before hitting the pattern def?

To the check, no. And a local variable(pattern definition) will always have an enclosing block of some sort, so we certainly could just check for a method defintion, and stop there. But, I generally avoid this sort of "optimization" since pitest normally fails on this sort of thing.

Reply:
I do not know why, but my brain not always work quick on do-while structure. I prefer while loop in this case.

Reply:
Not a big deal in this case, but I see the duplication of `getParent` and possibly an extra unnecessary condition check at the start of the loop and I think how to optimize it more. Hence the do-while version.

Reply:
> On second thought, I might be able to create a case to make this fail.

I am assuming this item is still TODO until told otherwise.

Reply:
>On second thought, I might be able to create a case to make this fail.

Disregard this.

Replaced while with do-while.

Reply:
So I saw the update, but I am still not really seeing a conclusion to this item:
> It doesn't matter if we pass any other kind of def before hitting the pattern def?

For example, it doesn't matter if we pass through an anonymous class to get to the pattern def or patterns are such that we are either in it or not for what we stop on.

Reply:
> anonymous class to get to the pattern def 

It is not possible to declare an anonymous class within a pattern definition.

>  patterns are such that we are either in it or not for what we stop on.

Please restate this, I am not sure that I understand what you are asking.

Reply:
Reminder. This is fix for exception, it should merged sooner. If we think there is some false positive, it is better to address in separate issue.

Reply:
I will approve the PR, but we have seen cases before where we either go too deep in the tree or go past too many key nodes when looking for "something" and we are accidentally pulling in nodes wrongly or passing things that should have stopped us. `branchContains` is one good example. https://github.com/checkstyle/checkstyle/issues/5124#issue-258583861 details the specifics of how it was going to far and wasn't be used carefully enough.

Whenever I see a loop where we are scanning up or down uncontrolled, I get concerned. I don't know the new Java syntax well, so I can't explain a possible situation to prompt a more meaningful conversation. I am just making sure this won't be an issue later.

Reply:
Condition is updated to exit on pattern def or expression node.